data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/handlers/beacon/GetStateTest.java

```diff
@@ -198,6 +198,6 @@ public void shouldReturnNotFoundWhenQueryByMissingSlot() throws Exception {
 
     handler.handle(context);
 
-    verify(context).status(SC_NOT_FOUND);
```

Comment:
as long as the slot is historical this is probably ok, might need another test case showing a future slot is not setting SC_GONE

Reply:
Updated to set the status based on whether slot is finalized.  If its finalized, we'll set the status to 410 GONE, if non-final we set it to 404 NOT_FOUND.  Added tests `shouldHandleMissingStateAtFinalizedSlot` and `shouldHandleMissingStateAtNonFinalSlot`.